### PR TITLE
Fix argument processing issues in CLI tools

### LINKED ISSUE #1992: CLI argument processing issues
# -h switch processing

Both CLI tools don't recognize short version of help command line switch `-h`, despite
it is being listed as acceptable
```
  -h, --help             This help message.
```

## Steps to Reproduce
```
rnpkeys -h
rnp -h
```
## Expected Behavior

Help message should be printed without error message about unrecognized option '-h'.

## Actual Behavior
```
rnpkeys: invalid option -- 'h'
...
rnp: invalid option -- 'h'
```

# Unnecessary loop in rnpkeys processing of --help and --version options

If multiple arguments are passed to rnpkeys CLI, it processes them in a loop using specified command.
https://github.com/rnpgp/rnp/blob/6d89b42b3e8680799feda7847ac88d619512ddc0/src/rnpkeys/main.cpp#L122
It is intended to process multiple files in a single run, but it leads to unexpected side effect when output for --help and --version commands could be printed multiple times.

## Steps to Reproduce
```
rnpkeys --help file1 file2 file3
rnpkeys --version file1 file2 file3
```
## Expected Behavior

Help and version messages should be printed only once.

## Actual Behavior
```
rnp 0.16.2+git20230211.df06da0
Ribose Inc. <rnpgp@ribose.com>
Backend: Botan
Backend version: 2.19.2
Supported algorithms:
Public key:  RSA, ELGAMAL, DSA, ECDH, ECDSA, EDDSA, SM2
Encryption:  IDEA, TRIPLEDES, CAST5, BLOWFISH, AES128, AES192, AES256, TWOFISH, CAMELLIA128, CAMELLIA192, CAMELLIA256, SM4
AEAD:  None, EAX, OCB
Key protection:  CFB
Hash:  MD5, SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224, SHA3-256, SHA3-512, SM3
Compression:  Uncompressed, ZIP, ZLIB, BZip2
Curves:  NIST P-256, NIST P-384, NIST P-521, Ed25519, Curve25519, brainpoolP256r1, brainpoolP384r1, brainpoolP512r1, secp256k1, SM2 P-256
Please report security issues at (https://www.rnpgp.org/feedback) and
general bugs at https://github.com/rnpgp/rnp/issues.
rnp 0.16.2+git20230211.df06da0
Ribose Inc. <rnpgp@ribose.com>
Backend: Botan
Backend version: 2.19.2
Supported algorithms:
Public key:  RSA, ELGAMAL, DSA, ECDH, ECDSA, EDDSA, SM2
Encryption:  IDEA, TRIPLEDES, CAST5, BLOWFISH, AES128, AES192, AES256, TWOFISH, CAMELLIA128, CAMELLIA192, CAMELLIA256, SM4
AEAD:  None, EAX, OCB
Key protection:  CFB
Hash:  MD5, SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224, SHA3-256, SHA3-512, SM3
Compression:  Uncompressed, ZIP, ZLIB, BZip2
Curves:  NIST P-256, NIST P-384, NIST P-521, Ed25519, Curve25519, brainpoolP256r1, brainpoolP384r1, brainpoolP512r1, secp256k1, SM2 P-256
Please report security issues at (https://www.rnpgp.org/feedback) and
general bugs at https://github.com/rnpgp/rnp/issues.
rnp 0.16.2+git20230211.df06da0
Ribose Inc. <rnpgp@ribose.com>
Backend: Botan
Backend version: 2.19.2
Supported algorithms:
Public key:  RSA, ELGAMAL, DSA, ECDH, ECDSA, EDDSA, SM2
Encryption:  IDEA, TRIPLEDES, CAST5, BLOWFISH, AES128, AES192, AES256, TWOFISH, CAMELLIA128, CAMELLIA192, CAMELLIA256, SM4
AEAD:  None, EAX, OCB
Key protection:  CFB
Hash:  MD5, SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224, SHA3-256, SHA3-512, SM3
Compression:  Uncompressed, ZIP, ZLIB, BZip2
Curves:  NIST P-256, NIST P-384, NIST P-521, Ed25519, Curve25519, brainpoolP256r1, brainpoolP384r1, brainpoolP512r1, secp256k1, SM2 P-256
Please report security issues at (https://www.rnpgp.org/feedback) and
general bugs at https://github.com/rnpgp/rnp/issues.

```

### PR DESCRIPTION
**Pull request by antonsviridenko**
Closes #1992

### CODE REVIEW

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
CodeCov wants a bit more coverage with tests though.

File: src/rnp/rnp.cpp [hunk 1]
```diff
@@ -626,6 +626,8 @@ rnp_main(int argc, char **argv)
                 print_usage(usage);
                 ret = EXIT_FAILURE;
                 goto finish;
+            case 'h':
+                [[fallthrough]];
```
Comment (ni4) [inline src/rnp/rnp.cpp:630]: Please cover this (and below) with some simple CLI test to make codecov happy.

**Review by ni4 — CHANGES_REQUESTED** [review]

**Comment by antonsviridenko** [comment]
I've discovered that both `rnp` and `rnpkeys` print help/usage message on stderr. I think it's wrong and should be printed on stdout instead, as it is not an error message.

**Comment by ni4** [comment]
> I've discovered that both `rnp` and `rnpkeys` print help/usage message on stderr. I think it's wrong and should be printed on stdout instead, as it is not an error message.

@antonsviridenko yeah, it would be more logical to dump help message to the stdout. Most likely it's leftover from the initial codebase.

File: src/rnp/rnp.cpp (hunk 1 above)
Comment (ronaldtse) [reply src/rnp/rnp.cpp:630]: Seems that CodeCov is really unhappy by dropping coverage from 80% to 60% 😉

**Comment by antonsviridenko** [comment]
Strange, all debian jobs failed for unknown reason at
```
        ret, out, _ = run_proc(RNPK, ['-h'])
        self.assertEqual(ret, 0)
```
ret = 1

**Comment by ni4** [comment]
@antonsviridenko I'll take a look, there should be a simple explanation (as well as to the problems with CI in the PR #1991 )

**Comment by ni4** [comment]
@antonsviridenko Here is the reason of debian failures:
```
2023-02-22T10:47:48.5713168Z Home directory '/home/rnpuser' does not exist or is not writable!
2023-02-22T10:47:48.5713271Z fatal: cannot set keystore info
```
Which is triggered by `rnpkeys_init` call in `main.cpp`.
Actually it would be more correct to not do any initialization in case of CMD_HELP, like this is done in `rnp`.

**Review by ni4 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**